<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigDialog</class>
 <widget class="QDialog" name="ConfigDialog">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>558</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Default font</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Font to use for newly created artists&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QFontComboBox" name="fontComboBox"/>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBox_fontSize"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The backward cache keeps frames which were already visited. Since GCD frames are merged into each Q and P frame, the size of this cache has to be at least 3 to make that possible without re-reading the GCD frames from the file.&lt;/p&gt;&lt;p&gt;The forward cache is used to pre-fetch the next frames, relative to the current position. This is done in a separate thread of execution, allowing you to work with the current frame while the next frames available fast.&lt;/p&gt;&lt;p&gt;Note: The cache is implemented at a deep level and always filled with all frames that are traversed, including those which are filtered out by Steamshovel.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="title">
      <string>Cache</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0" colspan="3">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Number of frames to keep in memory while browsing files.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Backward</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBox_backward">
        <property name="minimum">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSlider" name="slider_backward">
        <property name="minimum">
         <number>3</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Forward</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="spinBox_forward">
        <property name="frame">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QSlider" name="slider_forward">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Put the names of keys here that should be ignored when reading frames from disk. The typical use is to ignore the key &lt;span style=&quot; font-style:italic;&quot;&gt;I3Calibration&lt;/span&gt; which may take seconds to de-serialize, but is not used by any of the artists. Ignoring it makes your browsing experience significantly faster.&lt;/p&gt;&lt;p&gt;Any changes done here will not apply to files that are currently open. They will apply to the next opened files.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="title">
      <string>Performance</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Keys to ignore when frames are loaded. Separate by semi-colons.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="edit_ignore_keys">
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>No keys to ignore</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Convenience</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Choose a standard I3Geometry object to insert into the frame, if there is none available.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="textFormat">
         <enum>Qt::AutoText</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLineEdit" name="lineEdit_I3Geometry">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>No I3Geometry selected</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_I3Geometry">
          <property name="text">
           <string>Open file...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_I3Geometry_Clear">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Misc</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QCheckBox" name="checkBox_open_logviewer">
        <property name="text">
         <string>Automatically open log viewer if an error is reported</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>spinBox_backward</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slider_backward</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_forward</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slider_forward</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_forward</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_forward</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_backward</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_backward</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ConfigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>623</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ConfigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>403</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>233</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
